fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionExprDateTrunc.java

```diff
@@ -0,0 +1,224 @@
+
+    @Override
+    public void analyze(MTMVPartitionInfo mvPartitionInfo) throws AnalysisException {
+        if (!timeUnits.contains(this.timeUnit)) {
+            throw new AnalysisException(
+                    String.format("timeUnit not support: %s, only support: %s", this.timeUnit, timeUnits));
+        }
+        MTMVRelatedTableIf relatedTable = mvPartitionInfo.getRelatedTable();
+        PartitionType partitionType = relatedTable.getPartitionType();
+        if (partitionType == PartitionType.RANGE) {
+            Type partitionColumnType = MTMVPartitionUtil
+                    .getPartitionColumnType(mvPartitionInfo.getRelatedTable(), mvPartitionInfo.getRelatedCol());
+            if (!partitionColumnType.isDateType()) {
+                throw new AnalysisException(
+                        "partitionColumnType should be date/datetime "
+                                + "when PartitionType is range and expr is date_trunc");
+            }
+        }
+    }
+
+    @Override
+    public String getRollUpIdentity(PartitionKeyDesc partitionKeyDesc, Map<String, String> mvProperties)
+            throws AnalysisException {
+        String res = null;
+        Optional<String> dateFormat = getDateFormat(mvProperties);
+        List<List<PartitionValue>> inValues = partitionKeyDesc.getInValues();
+        for (int i = 0; i < inValues.size(); i++) {
+            // mtmv only support one partition column
+            PartitionValue partitionValue = inValues.get(i).get(0);
+            if (partitionValue.isNullPartition()) {
+                throw new AnalysisException("date trunc not support null partition value");
+            }
+            String identity = dateTrunc(partitionValue.getStringValue(), dateFormat, false).toString();
+            if (i == 0) {
+                res = identity;
+            } else {
+                if (!Objects.equals(res, identity)) {
+                    throw new AnalysisException(
+                            String.format("partition values not equal, res: %s, identity: %s", res,
+                                    identity));
+                }
+            }
+        }
+        return res;
+    }
+
+    private Optional<String> getDateFormat(Map<String, String> mvProperties) {
+        Optional<String> dateFormat =
+                StringUtils.isEmpty(mvProperties.get(PropertyAnalyzer.PROPERTIES_PARTITION_DATE_FORMAT))
+                        ? Optional.empty()
+                        : Optional.of(mvProperties.get(PropertyAnalyzer.PROPERTIES_PARTITION_DATE_FORMAT));
+        return dateFormat;
+    }
+
+    @Override
+    public PartitionKeyDesc generateRollUpPartitionKeyDesc(PartitionKeyDesc partitionKeyDesc,
+            MTMVPartitionInfo mvPartitionInfo) throws AnalysisException {
+        Type partitionColumnType = MTMVPartitionUtil
+                .getPartitionColumnType(mvPartitionInfo.getRelatedTable(), mvPartitionInfo.getRelatedCol());
+        // mtmv only support one partition column
```

Comment:
do we need add check here to avoid errors if we support more than one partition column and forget to do change here in future